src/data/set/finite.lean

```diff
@@ -594,6 +594,11 @@ theorem finite_mem_finset (s : finset α) : {a | a ∈ s}.finite := to_finite _
 lemma subsingleton.finite {s : set α} (h : s.subsingleton) : s.finite :=
 h.induction_on finite_empty finite_singleton
 
+lemma finite_preimage_inl_and_inr {s : set (α ⊕ β)} :
+  (sum.inl ⁻¹' s).finite ∧ (sum.inr ⁻¹' s).finite ↔ s.finite :=
```

Comment:
Do you think this should be called `finite_sum` instead, to match `locally_finite_sum` below?

Also, do we have the option version already?

Reply:
Then we should swap LHS with RHS. I'm not sure that we have an established naming (or LHS/RHS) convention for this kind of lemmas. I can start a topic on Zulip after ITP.

Reply:
Sorry, I meant to include "swap sides" in that comment.

Reply:
What is the status here?

Reply:
I've got covid and didn't start the discussion on Zulip.